### ISSUE TITLE
std:bad_alloc

### ISSUE THREAD

**Issue by gdkrmr** [hadley/dplyr#2323]
running the following code:

```
library(dplyr)
rnx <- data.frame(K = 1:999)
expand.grid(K = rnx$K,
            rnx = seq(0.1, 1, by = 0.1)) %>%
    dplyr::mutate(qnx = rnx2qnx(rnx), rnx_group = factor(K))
```

gives the following error:
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
```

I tried from inside an R session and a separate file with Rscript, both give the same error.

My computer:
```
R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.1 LT
dplyr_0.5.0
```

**Comment by krlmlr** [comment]
Thanks. What is `rnx2qnx`? Could you please provide a reproducible example?

Also, please double-check if this still occurs in the development version:

```r
devtools::install_github("hadley/dplyr")
```

**Comment by gdkrmr** [comment]
That was a reproducible example before I went to have lunch. I just had to copy the lines into the R repl or into a file and run it with Rscript, the function did not have to be in global namespace, the crash must have happen before lazy evaluation.

in the above example dplyr master gives the the expected error that it cannot see `rnx2qnx` and if I correct the bugs in my code it works as expected.

Now I reverted back to dplyr 0.5.0 and I cannot reproduce it any more. I will let you know I it appears again.
